src/client/refresh/gl1/gl1_main.c

```diff
@@ -1667,6 +1669,26 @@ RI_BeginFrame(float camera_separation)
 	glEnable(GL_ALPHA_TEST);
 	glColor4f(1, 1, 1, 1);
 
+	if (gl1_particle_square->modified)
+	{
+		gl1_particle_square->modified = false;
+
+		/* yamagi: GL_POINT_SMOOTH is not implemented by some
+		   OpenGL drivers, especially the crappy Mesa3D backends
+		   like i915.so. That the points are squares and not
+		   circles is not a problem by Quake II! */
+
+		switch ((int)gl1_particle_square->value)
+		{
+			default:
+				glDisable(GL_POINT_SMOOTH);
+				break;
+			case 0:
+				glEnable(GL_POINT_SMOOTH);
+				break;
+		}
+	}
```

Comment:
This block must be wrapped with `if (gl_config.pointparameters)`. Without it GPUs that don't support the pointparameter extension may run into problem. At `gl1_pointparameter 0` might be broken.

Reply:
also, why a switch for just two cases (instead of if/else)?